README.md
# SnowpiercerMod

Hello! This is my first mod for factorio. I made this mod because I really love the world Snowpiercer. Factorio seemed like the perfect platform for me to recreate the world.
This mod adds 2 trains, a reinforced freight car and a power cell to power the trains. Mod settings support changing the maximum speed of the train.

Yes, the mod is simple, but it was a good experience to create mods for this wonderful game. :) 

Snowpiercer world map: https://drive.google.com/file/d/13rh9lSJpdf8KnBa_DQx87H6Klry0xKh2/view?usp=sharing
I hope you will like it :)


changelog.txt
ver. 1.1.1
Date: 04/04/22
 Fix - Fixed loss of energy cell when mining locomotives

ver. 1.1.0
Date: 20/02/22
 Add engine sounds
 Add stop effect
 Fix - Defining additional mods and modloader
Thank you so much for 100 downloads!

ver. 1.0.1
Date: 05/02/22
 Fix - no mod recognition

prototypes/items.lua
data:extend(
{
    {
        type = "item-with-entity-data",
        name = "Snowpierser",
        subgroup = "train-transport",
        icon = "__Snowpierсer__/graphics/SnowpiercerLoc.png",
        order = "h[train-system]",
        icon_size = 64,
        icon_mipmaps = 4,
        place_result = "Snowpierser",
        stack_size = 1
    },
    {
        type = "item-with-entity-data",
        name = "BigElice",
        subgroup = "train-transport",
        icon = "__Snowpierсer__/graphics/BigAliceLo.png",
        order = "h[train-system]",
        icon_size = 64,
        icon_mipmaps = 4,
        place_result = "BigElice",
        stack_size = 1
    },
    {
        type = "item-with-entity-data",
        name = "Snowcar",
        subgroup = "train-transport",
        icon = "__Snowpierсer__/graphics/Snowcar.png",
        order = "h[train-system]",
        icon_size = 32,
        place_result = "Snowcar",
        stack_size = 10
    },
    {
        type = "item",
        icon = "__Snowpierсer__/graphics/Energy-cell.png",
        icon_size = 32,
        name = "perpetual-motion-cell",
        fuel_category = "chemical",
        fuel_value = "9999999GJ",
        subgroup = "raw-material",
        order = "h[battery]",
        stack_size = 1
    }
}
)

prototypes/recipes.lua
data:extend(
    {
        --Рассекающий
        {
            type = "recipe",
            name = "Snowpierser",
            energy_required = 0.5,
            ingredients = {
                {"steel-plate", 200},
                {"iron-gear-wheel",500},
                {"fusion-reactor-equipment", 1},
                {"electronic-circuit",200},
                {"advanced-circuit",100},
                {"processing-unit",20}
            },
            result = "Snowpierser",
            enabled = true,
            subgroup = "train-transport",
            icon = "__Snowpierсer__/graphics/SnowpiercerLoc.png",
            order = "aaa",
            icon_size = 64
        },
        --Большая Элис
        {
            type = "recipe",
            name = "BigElice",
            energy_required = 0.5,
            ingredients = {
                {"steel-plate", 100},
                {"iron-gear-wheel",250},
                {"fusion-reactor-equipment", 1},
                {"electronic-circuit",200},
                {"advanced-circuit",100},
                {"processing-unit",20}
            },
            result = "BigElice",
            enabled = true,
            subgroup = "train-transport",
            icon = "__Snowpierсer__/graphics/BigAliceLo.png",
            order = "aaa",
            icon_size = 32
        },
        --Грузовые вагоны
        {
            type = "recipe",
            name = "Snowcar",
            energy_required = 0.5,
            ingredients = {
                {"iron-plate", 100},
                {"steel-plate", 50},
                {"iron-gear-wheel",100}
            },
            result = "Snowcar",
            enabled = true,
            subgroup = "train-transport",
            icon = "__Snowpierсer__/graphics/Snowcar.png",
            order = "aaa",
            icon_size = 32
        },
        --Энергоячейка
        {
            type = "recipe",
            name = "perpetual-motion-cell",
            energy_required = 0.5,
            ingredients = {
                {"uranium-fuel-cell", 10},
                {"nuclear-reactor", 1}
            },
            result = "perpetual-motion-cell",
            enabled = true,
            subgroup = "raw-material",
            icon = "__Snowpierсer__/graphics/Energy-cell.png",
            order = "aaa",
            icon_size = 32
        },
    }
)

prototypes/entities.lua
-- Max_Speed = 216 * Max_speed_coefficient
local Max_speed_coefficient = settings.startup["Snowpiercer-speed-coefficient"].value

--Прототип Рассекающего
local Snowpiercer = table.deepcopy(data.raw["locomotive"]["locomotive"])
Snowpiercer.name = "Snowpierser"
Snowpiercer.minable.result = "Snowpierser"
Snowpiercer.burner.smoke = nil
Snowpiercer.burner.fuel_inventory_size = 1
Snowpiercer.max_power = "50MW"
Snowpiercer.max_speed = Max_speed_coefficient
Snowpiercer.reversing_power_modifier = 1
Snowpiercer.weight = 3000
Snowpiercer.color = {0.5, 0.5, 0.5, 1}
Snowpiercer.LightDefinition = {intensity = 1, size = 30, color = {r = 1, g =0.5}}
Snowpiercer.alert_when_damaged= true
Snowpiercer.flag = {"rotatable"}
Snowpiercer.working_sound =
{
  sound =
  {
    filename = "__Snowpiercer__/sound/working-sound.ogg",
    volume = 1.0
  },

  activate_sound =
  {
    filename = "__Snowpiercer__/sound/engine-start.ogg",
    volume = 1.0
  },
  deactivate_sound = {
    filename = "__Snowpiercer__/sound/stop_engeen.ogg",
    volume = 2.0
  }
}
Snowpiercer.stop_trigger =
{
  -- left side
  {
    type = "create-trivial-smoke",
    repeat_count = 125,
    smoke_name = "smoke-train-stop",
    initial_height = 0,
    -- smoke goes to the left
    speed = {-0.03, 0},
    speed_multiplier = 0.75,
    speed_multiplier_deviation = 1.1,
    offset_deviation = {{-0.75, -2.7}, {-0.3, 2.7}}
  },
  -- right side
  {
    type = "create-trivial-smoke",
    repeat_count = 125,
    smoke_name = "smoke-train-stop",
    initial_height = 0,
    -- smoke goes to the right
    speed = {0.03, 0},
    speed_multiplier = 0.75,
    speed_multiplier_deviation = 1.1,
    offset_deviation = {{0.3, -2.7}, {0.75, 2.7}}
  },
  {
    type = "play-sound",
    sound =
    {
      {
        filename = "__base__/sound/train-breaks.ogg",
        volume = 0.3
      }
    } 
  },
  {
    type = "play-sound",
    sound =
    {
      {
        filename = "__base__/sound/train-brake-screech.ogg",
        volume = 0.3
      }
    } 
  }
}

--Прототип Большой Элис
local BigElice = table.deepcopy(data.raw["locomotive"]["locomotive"])
BigElice.name = "BigElice"
BigElice.minable.result = "BigElice"
BigElice.burner.smoke = nil
BigElice.burner.fuel_inventory_size = 1
BigElice.max_power = "30MW"
BigElice.max_speed = Max_speed_coefficient
BigElice.braking_force = 20
BigElice.reversing_power_modifier = 5
BigElice.weight = 5000
BigElice.color = {0.5, 0.5, 0.5, 1}
BigElice.LightDefinition = {intensity = 1, size = 30, color = {r = 1, g =0.5}}
BigElice.alert_when_damaged = true
BigElice.flag = {"rotatable"}
BigElice.working_sound =
{
  sound =
  {
    filename = "__Snowpiercer__/sound/working-sound.ogg",
    volume = 1.0
  },

  activate_sound =
  {
    filename = "__Snowpiercer__/sound/engine-start.ogg",
    volume = 1.0
  },
  deactivate_sound = {
    filename = "__Snowpiercer__/sound/stop_engeen.ogg",
    volume = 2.0
  }
}
BigElice.stop_trigger =
{
  -- left side
  {
    type = "create-trivial-smoke",
    repeat_count = 125,
    smoke_name = "smoke-train-stop",
    initial_height = 0,
    -- smoke goes to the left
    speed = {-0.03, 0},
    speed_multiplier = 0.75,
    speed_multiplier_deviation = 1.1,
    offset_deviation = {{-0.75, -2.7}, {-0.3, 2.7}}
  },
  -- right side
  {
    type = "create-trivial-smoke",
    repeat_count = 125,
    smoke_name = "smoke-train-stop",
    initial_height = 0,
    -- smoke goes to the right
    speed = {0.03, 0},
    speed_multiplier = 0.75,
    speed_multiplier_deviation = 1.1,
    offset_deviation = {{0.3, -2.7}, {0.75, 2.7}}
  },
  {
    type = "play-sound",
    sound =
    {
      {
        filename = "__base__/sound/train-breaks.ogg",
        volume = 0.3
      }
    } 
  }
}

--Прототип грузовых вагонов
local SCargowagon = table.deepcopy(data.raw["cargo-wagon"]["cargo-wagon"])
SCargowagon.name = "Snowcar"
SCargowagon.inventory_size = 100
SCargowagon.color = {0.5,0.5,0.5,1}
SCargowagon.weight = 2000
SCargowagon.max_speed = Max_speed_coefficient
SCargowagon.braking_force = 4
SCargowagon.minable.result = "Snowcar"
SCargowagon.alert_when_damaged = true
SCargowagon.max_health = 1000
SCargowagon.resistances =
{
  {
    type = "physical",
    decrease = 6,
    percent = 50
  },
  {
    type = "explosion",
    decrease = 20,
    percent = 50
  },
  {
    type = "acid",
    decrease = 3,
    percent = 50
  }
}
SCargowagon.stop_trigger =
{
  -- left side
  {
    type = "create-trivial-smoke",
    repeat_count = 125,
    smoke_name = "smoke-train-stop",
    initial_height = 0,
    -- smoke goes to the left
    speed = {-0.03, 0},
    speed_multiplier = 0.75,
    speed_multiplier_deviation = 1.1,
    offset_deviation = {{-0.75, -2.7}, {-0.3, 2.7}}
  },
  -- right side
  {
    type = "create-trivial-smoke",
    repeat_count = 125,
    smoke_name = "smoke-train-stop",
    initial_height = 0,
    -- smoke goes to the right
    speed = {0.03, 0},
    speed_multiplier = 0.75,
    speed_multiplier_deviation = 1.1,
    offset_deviation = {{0.3, -2.7}, {0.75, 2.7}}
  },
  {
    type = "play-sound",
    sound =
    {
      {
        filename = "__base__/sound/train-breaks.ogg",
        volume = 0.3
      }
    } 
  }
}

data:extend(
    {Snowpiercer, BigElice, SCargowagon}
)